src/NServiceBus.Core/Transports/Learning/DirectoryBasedTransaction.cs

```diff
@@ -59,11 +71,11 @@ public Task Enlist(string messagePath, string messageContents)
             return AsyncFile.WriteText(txPath, messageContents);
         }
 
-        public void Complete()
+        public Task<bool> Complete()
```

Comment:
I don't see an issue to switch over to Task based API once we need it, it's all private, isn't it?